README.md

```diff
@@ -1375,12 +1375,77 @@ Just like how a `class` may extend its properties by inheriting `class` or `abst
     age: 74,
     firstName: 'An Old Guy',
     greet(lastName: string) {
-      console.log(`Hello Sr. ${lastName}.`);
+      console.log(`Hello Sr. ${lastName}.`); // Would print: "Hello Sr. An Old Guy"
     }
   }
 ```
 
-As you can see, the `oldPerson` will be restricted to the `AgePerson` type structure, which inherits all of `NamedPerson` properties.
+As you can see, the `oldPerson` will be restricted to the `AgedPerson` type structure, which inherits all of `NamedPerson` properties.
+
+Now, instead of overwriting a property that could be `undefined` previously, this time let's try to add new properties on top of `AgedPerson` by extending it yet again to a new interface:
+
+```ts
+  interface NamedPerson {
+    firstName: string;
+    age?: number;
+    [propName: string]: any;
+    greet(lastName: string): void;
+  }
+
+  interface AgedPerson extends NamedPerson {
+    age: number;
+  }
+
+  interface AthleticPerson extends AgedPerson {
+    isAthlete: true
+    isSuccessful?: boolean
+  }
+
+  const athleticPerson: AthleticPerson = {
+    age: 29,
+    firstName: 'An Old Guy',
+    isAthlete: true,
+    greet(lastName: string) {
+      console.log(`Hello Sr. ${lastName}.`); // Would print: "Hello Sr. An Old Guy"
+    }
+  }
+```
+
+Notice, how the `AthleticPerson` interface has the properties of not just `AgedPerson`, but of `NamedPerson`, plus of course its own properties. This is a process that can be repeated over, and over, so long as the interface structures are compatible.
+
+The example above can also be written as:
```

Comment:
I'd remove this entire section, it's not wrong but hasn't anything to do with the core idea of `extends`

Reply:
Isn't the core idea of extending interfaces, is to create a combination of all of the interfaces? Basically copying one into another. I think it's necessary, but can be reworded to deliver the point in a better way. What do you think?

Reply:
The core idea of interfaces is creating a contract for how to exchange data with the external world e.g. an API of a third party library or an entire suite of such libs (eg firebase).

The core idea of `extends` is composition of said contracts (named interfaces in Typescript). It's not quite copying, more merging were shadowing a property becomes a possibility (as you've shown with `age` vs `age?`.

Given that I'd agree, you can leave things but then it needs heavy rewording imo, maybe even start over and take the official docs text/examples as a pointer.